README.rst

```diff
@@ -95,6 +95,67 @@ View your user::
 
    linode-cli profile view
 
+Kubernetes Deployment Plugin
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A plugin is included that allows you to deploy a Kubernetes cluster on Linode!
+
+This Kubernetes cluster is integrated with Linode in a number of ways:
+
+   * When you deploy a LoadBalancer-type service a Linode NodeBalancer will be
+     automatically created and managed for the Pods backing that service.
+   * When PersistentVolumes are created, those volumes will be Linode Block
+     Storage volumes. These are also automatically managed with the lifecycle of the
+     PersistentVolume resource.
+   * Nodes have the appropriate InternalIP, ExternalIP and ProviderID fields,
+     meaning that CNI and other controllers can take advantage of these fields
+     for the sake of NetworkPolicy and other Kubernetes features.
+   * Nodes are labeled with the Linode Region and Linode Type, which can also
+     be used by controllers for the purposes of scheduling
+   * The metrics-server is installed, allowing you to use `kubectl top`
+
+The following is the help message for the command
+
+   $ linode-cli k8s-alpha create --help
+   usage: k8s-alpha create [-h] [--node-type TYPE] [--nodes COUNT]
+                           [--master-type TYPE] [--region REGION]
+                           [--ssh-private-key KEYPATH] [--ssh-public-key KEYPATH]
+                           NAME
+
+   positional arguments:
+     NAME                  A name for the cluster.
+
+   optional arguments:
+     -h, --help            show this help message and exit
+     --node-type TYPE      Linode Type ID for cluster Nodes as retrieved with
+                           `linode-cli linodes types`. (default "g6-standard-2")
+     --nodes COUNT         The number of Linodes to deploy as Nodes in the
+                           cluster. (default 3)
+     --master-type TYPE    Linode Type ID for cluster Master Nodes as retrieved
+                           with `linode-cli linodes types`. (default
+                           "g6-standard-2")
+     --region REGION       The Linode Region ID in which to deploy the cluster as
+                           retrieved with `linode-cli regions list`. (default
+                           "us-west")
+     --ssh-private-key KEYPATH
+                           The path to your private key file which will be used
+                           to access Nodes
+     --ssh-public-key KEYPATH
+                           The path to your public key file which will be used to
+                           access Nodes
+
+Here's an example use of the command, creating a cluster with 6 2GB Linodes as
+the Nodes.
+
+   linode-cli k8s-alpha create mycluster77 --node-type g6-standard-1 --nodes 6 --master-type g6-standard-4 --region us-east --ssh-private-key $HOME/.ssh/id_rsa --ssh-public-key $HOME/.ssh/id_rsa.pub
```

Comment:
This doesn't send your key - it just instructs Terraform on which key to use when provisioning the nodes.  Terraform will deploy the instances with the public key specified, but it wants to know which private key it will need to use when it uses ssh to reach the nodes.

Reply:
I personally have avoided using the personal key, deferring to the ssh agent support.  But I could see wanting to use different key, not contained in your agent, for this purpose.

Maybe you are running this headless (without an agent), or want to use a unique SSH key pair per cluster.

Reply:
It may be worth mentioning this in the documentation, FWIW; this is the second time I've seen this brought up.

Reply:
🙃 TIL

Reply:
I'm going to do some testing and see how smart Terraform's agent is about inferring the location of a privkey from a public key path. Maybe this isn't needed. If it is, it will come with an explanation.